go/cmd/launcher/main.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2019 Red Hat, Inc.
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"github.com/virtblocks/virtblocks/go/pkg/command"
+	"github.com/virtblocks/virtblocks/go/pkg/devices"
+	"github.com/virtblocks/virtblocks/go/pkg/vm"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+func processFlags() (disk string, cpus uint, mem uint, err error) {
+	flag.StringVar(&disk, "disk", "", "path to disk image (qcow2)")
+	flag.UintVar(&cpus, "cpus", 1, "number of CPUs")
+	flag.UintVar(&mem, "mem", 128, "amount of memory (MiB)")
+
+	flag.Parse()
+
+	if disk == "" {
+		err = errors.New("need a disk")
+		return
+	}
+	if filepath.Ext(disk) != ".qcow2" {
+		err = errors.New("need disk in qcow2 format")
+		return
+	}
+	if cpus < 1 {
+		err = errors.New("need at least one CPU")
+		return
+	}
+	if mem < 64 {
+		err = errors.New("need more memory")
+		return
+	}
+
+	return
+}
+
+func main() {
+	diskPath, cpus, mem, err := processFlags()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error processing flags: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Derive the path of the socket used for serial console communication
+	// from the path of the disk image
+	serialSockPath := fmt.Sprintf("%s%s", strings.TrimSuffix(diskPath, "qcow2"), "serial-sock")
+
+	vm := vm.NewDescription()
+	vm.SetCpus(cpus).SetMemory(mem)
+	vm.SetDisk(devices.NewDisk().SetFilename(diskPath))
+	vm.SetSerial(devices.NewSerial().SetPath(serialSockPath))
+
+	qemuArgs, err := vm.QemuCommandLine()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error generating QEMU command: %v\n", err)
+		os.Exit(1)
+	}
+
+	qemu := command.NewCommand(qemuArgs[0])
```

Comment:
I am still unsure about the `command` package ;)

Reply:
I know you're not exactly a fan :)

I think that API might potentially become more interesting once we start dealing with FD passing (which I hope I'm going to be able to work on this week). Or maybe not! We'll see.

In the context of the silly demo, I think using it is perfectly fine... It's not like we're making the QEMU command line API depend on it, which would of course require much more consideration.

If we end up deciding we don't want to keep this API around, then it will mean it hasn't provided any value and porting to `exec.Cmd` should be trivial.